### PR TITLE
kubeadm: don't mount /etc/pki for apiserver and KCM

### LINKED ISSUE #1665: caCertsExtraVolumePaths should point to the correct distro paths
/kind bug
/area security
/sig cluster-lifecycle

## Versions

**kubeadm version** (use `kubeadm version`): 1.15

## What happened?

In https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/controlplane/volumes.go#L42, caCertsExtraVolumePaths is an array of common system CA certificate paths. However, for RHEL/Fedora-based systems, the definition of `/etc/pki` is a higher level directory that also grants access to private keys for daemons following the Fedora packaging standards.

## What you expected to happen?

The true CA certificate bundle path for openssl on Fedora-derived systems is `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`. 
`/etc/pki/ca-trust/extracted/pem/` can also be used for a directory of PEM files, as per the README in that directory:

```
This directory /etc/pki/ca-trust/extracted/pem/ contains
CA certificate bundle files which are automatically created
based on the information found in the
/usr/share/pki/ca-trust-source/ and /etc/pki/ca-trust/source/
directories.

All files are in the BEGIN/END CERTIFICATE file format,
as described in the x509(1) manual page.

Distrust information cannot be represented in this file format,
and distrusted certificates are missing from these files.

If your application isn't able to load the PKCS#11 module p11-kit-trust.so,
then you can use these files in your application to load a list of global
root CA certificates.
```

## How to reproduce it (as minimally and precisely as possible)?
N/A

## Anything else we need to know?

The certificate bundle is derived from the ca-trust store:

> In Red Hat Enterprise Linux, the consolidated system-wide trust store is located in the /etc/pki/ca-trust/ and /usr/share/pki/ca-trust-source/ directories. The trust settings in /usr/share/pki/ca-trust-source/ are processed with lower priority than settings in /etc/pki/ca-trust/.

openssl will then consume them from `/etc/pki/tls/certs/ca-bundle.crt`

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/using-shared-system-certificates_security-hardening

### PR DESCRIPTION
**Pull request by neolit123**
#### What type of PR is this?

/kind bug cleanup

#### What this PR does / why we need it:

According to kubeadm repo ticket 1665, /etc/pki
can contain subdirectories with private keys on some distros.
Avoid mounting the entire /etc/pki and mount /etc/pki/ca-trust
and /etc/pki/tls/certs instead. These directories are mounted
as an extra locations which can be used to search
for additional system CAs.

i can't find additional information about these private keys, but mounting /etc/pki/ca-trust, /etc/pki/tls/certs  and /etc/ca-certificates from etc should suffice. we are already mounting other locations too.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubeadm/issues/1665

#### Does this PR introduce a user-facing change?

```release-note
kubeadm: don't mount /etc/pki in kube-apisever and kube-controller-manager pods as an additional Linux system CA location. Mount /etc/pki/ca-trust and /etc/pki/tls/certs instead. /etc/ca-certificate, /usr/share/ca-certificates, /usr/local/share/ca-certificates and /etc/ssl/certs continue to be mounted.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/phases/controlplane/volumes.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ const (
 // caCertsExtraVolumePaths specifies the paths that can be conditionally mounted into the apiserver and controller-manager containers
 // as /etc/ssl/certs might be or contain a symlink to them. It's a variable since it may be changed in unit testing. This var MUST
 // NOT be changed in normal codepaths during runtime.
-var caCertsExtraVolumePaths = []string{"/etc/pki", "/usr/share/ca-certificates", "/usr/local/share/ca-certificates", "/etc/ca-certificates"}
+var caCertsExtraVolumePaths = []string{"/etc/pki/ca-trust", "/etc/ca-certificates", "/usr/share/ca-certificates", "/usr/local/share/ca-certificates"}
```
[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/phases/controlplane/volumes.go:43]: seems so, missed it in the discussion on https://github.com/kubernetes/kubeadm/issues/1665

[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/phases/controlplane/volumes.go]: updated.

[the remark this answers is not in the document]
Comment (SataQiu) [reply cmd/kubeadm/app/phases/controlplane/volumes.go]: /lgtm
/hold feel free to cancel hold

**Comment by neolit123** [comment]
/hold cancel

let's update if this regresses some linux distros.
either way mounting the root /etc/pki seems wrong.

**Pull request merged**